src/main/java/org/motechproject/mots/service/UserService.java

```diff
@@ -38,11 +39,29 @@ public User getUser(UUID id) {
         new EntityNotFoundException("User with id: {0} not found", id.toString()));
   }
 
+  @PreAuthorize(RoleNames.HAS_MANAGE_USERS_ROLE)
```

Comment:
if you use Manage users role here, only admin will be able to change his password, remove this